### PR TITLE
cpu/stm32_common: make RTT_FREQUENCY configurable

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR is split from https://github.com/RIOT-OS/RIOT/pull/13874. It unifies `rtt` configuration for `stm32%` and make the `RTT_FREQUENCY` configurable.

As in #13874 max and min values for the RTT are documented.

I also had to set `RTT_FREQUENCY ` for `stm32%` board to avoid `TICKS_TO_WAIT` overflowing, I kept the previous value for it.

### Testing procedure

- `tests/periph_rtt` should still pass on `stm32%` boards with different frequencies

`BOARD=b-l072z-lrwan1make -C tests/periph_rtt flash test --no-print-directory -j3`
```
main(): This is RIOT! (Version: 2020.07-devel-110-g66faf-pr_stm32_rtt)

RIOT RTT low-level driver test
This test will display 'Hello' every 5 seconds

Initializing the RTT driver
RTT now: 0
Setting initial alarm to now + 5 s (5120)
Done setting up the RTT, wait for many Hellos
Hello
Hello
Hello
Hello
Hello
```

`CFLAGS+=-DRTT_FREQUENCY=2048 BOARD=iotlab-m3 make -C tests/periph_rtt clean flash test --no-print-directory -j3`

```
main(): This is RIOT! (Version: 2020.07-devel-110-g66faf-pr_stm32_rtt)

RIOT RTT low-level driver test
This test will display 'Hello' every 5 seconds

Initializing the RTT driver
RTT now: 0
Setting initial alarm to now + 5 s (10240)
Done setting up the RTT, wait for many Hellos
Hello
Hello
Hello
Hello
Hello
```

### Issues/PRs references

Split from https://github.com/RIOT-OS/RIOT/pull/13874

### CODE REVIEW

File: boards/common/stm32/include/cfg_rtt_default.h [hunk 1]
```diff
@@ -31,8 +31,21 @@ extern "C" {
...
  * @{
  */
-#define RTT_FREQUENCY       (1024U)             /* 32768 / 2^n */
-#define RTT_MAX_VALUE       (0x0000ffff)        /* 16-bit timer */
+#ifdef LPTIM1
+#ifdef CLOCK_LSE
+#define RTT_CLOCK_FREQUENCY (32768U)                    /* in Hz */
+#else
+#define RTT_CLOCK_FREQUENCY (CLOCK_LSI)                 /* in Hz */
+#endif
```
Comment (fjmolinas) [inline boards/common/stm32/include/cfg_rtt_default.h:39]: Ups this doesn't work for boards defining `CLOCK_LSE` after loading `cfg_rtt_default.h`. I could fix it by:

- moving clock configuration to a `boards/common/stm32/include/x/cfg_clock_x_x_x.h` configuration file.
- including `cfg_rtt_default.h` after the clock definitions for those boards.
- Add `CLOCK_LSX` variable that would be set to `CLOCK_LSE` or `CLOCK_LSI` depending on the configuration, the precompiler could then resolve `#define RTT_CLOCK_FREQUENCY (CLOCL_LSX) `

Option two requires the smallest amount of changes.

Comment (fjmolinas) [reply boards/common/stm32/include/cfg_rtt_default.h]: Simply removed this part for, `CLOCK_LSI` based `rtt` is not really supported currently

File: tests/periph_rtt/Makefile [hunk 2]
```diff
@@ -6,3 +6,11 @@ FEATURES_REQUIRED = periph_rtt
 DISABLE_MODULE += periph_init_rtt
 
 include $(RIOTBASE)/Makefile.include
+
+# Put board specific dependencies here
+ifneq (,$(filter-out stm32f1,$(filter stm32%,$(CPU))))
+  # all stm32% but stm32f1 RTT are based on a 16 bit LPTIM, if using the default
+  # 32768KHz configuration TICKS_TO_WAIT will overflow
+  RTT_FREQUENCY ?= 1024
+  CFLAGS += -DRTT_FREQUENCY=$(RTT_FREQUENCY)
```
Comment (benpicco) [inline tests/periph_rtt/Makefile:15]: This will break if the board already defined `RTT_FREQUENCY`.

You could use `MOCK_RTT_FREQUENCY`, but that would be a bit of a hack.

Comment (fjmolinas) [reply tests/periph_rtt/Makefile:15]: > This will break if the board already defined RTT_FREQUENCY.
> 
> You could use MOCK_RTT_FREQUENCY, but that would be a bit of a hack.

Yes, that why I filtered to only allow `stm32` CPU's, with this PR they would all use the same configuration that has the `ifndef` logic to allow it.

Comment (fjmolinas) [reply tests/periph_rtt/Makefile:15]: Didn't see the `nucleo-f413zh`, I will address that.

**Comment by fjmolinas** [comment]
@aabadie @benpicco ping!

File: boards/common/stm32/include/cfg_rtt_default.h [hunk 3]
```diff
@@ -31,8 +31,17 @@ extern "C" {
  * On the STM32Lx platforms, we always utilize the LPTIM1.
  * @{
  */
-#define RTT_FREQUENCY       (1024U)             /* 32768 / 2^n */
-#define RTT_MAX_VALUE       (0x0000ffff)        /* 16-bit timer */
+#ifdef LPTIM1
```
Comment (aabadie) [inline boards/common/stm32/include/cfg_rtt_default.h:34]: Why is this needed ?

Comment (fjmolinas) [reply boards/common/stm32/include/cfg_rtt_default.h:34]: I think it was needed before, vut I fixed that for the `stm32f1` pr changing `rtt`, probably not needed anymore, let me re-check!

Comment (fjmolinas) [reply boards/common/stm32/include/cfg_rtt_default.h]: lets seee if not needed anymore..

**Comment by fjmolinas** [comment]
All green here @aabadie !

**Comment by fjmolinas** [comment]
> Changes are also good to me even if I would have preferred to keep the test application platform agnostic. Anyway I won't block for that.
> 
> So ACK

Thanks for the review!
